src/main/scala/org/ergoplatform/nodeView/history/extra/ExtraIndexer.scala

```diff
@@ -457,6 +460,6 @@ object ExtraIndexer {
   * Base trait for all additional indexes made by ExtraIndexer
   */
 trait ExtraIndex {
-  def id: ModifierId = bytesToId(serializedId)
-  def serializedId: Array[Byte]
+  lazy val id: ModifierId = bytesToId(serializedId)
+  def serializedId: Array[Byte] = fastIdToBytes(id)
```

Comment:
Looks like mutually recursive definitions.

Reply:
Yes, but one of them is always overidden

Reply:
@jellymlg, I assume it is serializedId? Then why not to make it abstract and let the compiler check it is indeed overriden?

Reply:
IndexedErgoAddress overrides id
IndexedErgoTransaction overrides id
IndexedToken overrides id
IndexedErgoBox overrides serializedId
NumericTxIndex overrides serializedId
NumericBoxIndex overrides serializedId